subiquity/common/filesystem/gaps.py

```diff
@@ -41,6 +41,26 @@ class Gap:
     def id(self):
         return 'gap-' + self.device.id
 
+    def copy(self):
+        return Gap(device=self.device,
+                   offset=self.offset,
+                   size=self.size,
+                   in_extended=self.in_extended)
+
+    def split(self, size):
```

Comment:
Should this care about alignment issues at all? Or is that the callers responsibility? (I say probably the callers responsibility?)

Reply:
Good question.  Since the `gaps.find_disk_gaps` functions are alignment aware, probably other gaps operations should be also.

The `align_up` in `FilesystemModel.add_partition()` concerns me, I think the right place is for that to happen a bit earlier, and maybe gaps help with that.

Will consider for a future branch.